R/utils-test.R

```diff
@@ -0,0 +1,31 @@
+#' Construct example financial data
+#'
+#' @param ... Passed to [tibble()].
+#'
+#' @return A [tibble()]
+#' @keywords internal
+#' @export
+#'
+#' @examples
+#' example_financial()
+example_financial <- function(bank_id = "a",
+                              amount_total = 10,
+                              company_name = "b",
+                              emission_profile = "medium",
+                              benchmark = "c",
+                              equal_weight_finance = 10,
+                              worst_case_finance = 10,
+                              best_case_finance = 10,
+                              ...) {
```

Comment:
Great! This is exactly how I would do it. 

To avoid littering your tests with brittle column names, you may want to support creating example datasets with your `example_*()` function in combination with stable interfaces that hide those brittle column names. I show this in the reprex. Let me know if it's unclear.

<details/>
<summary/>
reprex
</summary>

``` r
  library(rlang, warn.conflicts = FALSE)
  library(tibble, warn.conflicts = FALSE)

  # R/example_datasets.R -------------------------------------------------------
  
  # Idea used in tiltIndicator::example_companies() and friends
  # You need to add `:=` with this: `usethis::use_import_from("rlang", ":=")`
  example_financial <- function (...)  {
    new <- tibble(...)
    old <- as.list(example_financial_impl())
    add <- setdiff(names(old), names(new))
    as_tibble(append(new, old[add]))
  }
  # This is exactly what you wrote but without the dots `...` 
  example_financial_impl <- function(bank_id = 1, other = 1) {
    tibble(
      bank_id = bank_id,
      other = other
    )
  }
  
  # This file mentions brittle names, so if it changes you will also need to change
  # column names everywhere else, unless everywhere you access those brittle
  # names through stable interfaces that YOU control. So it seems like a good
  # place to map those brittle names to your stable interfaces
  get_id <- function() "bank_id"
  get_europages_product <- function() "ep_product"
  
  
  
  # Usage -----------------------------------------------------------------------
  
  # This version of `example_financial()` supports the basic behaviour ...
  example_financial(bank_id = 1:2)
#> # A tibble: 2 × 2
#>   bank_id other
#>     <int> <dbl>
#> 1       1     1
#> 2       2     1
  example_financial(bank_id = NULL)
#> # A tibble: 1 × 2
#>   bank_id other
#>     <dbl> <dbl>
#> 1       1     1
  example_financial(new = 1)
#> # A tibble: 1 × 3
#>     new bank_id other
#>   <dbl>   <dbl> <dbl>
#> 1     1       1     1
  
  # But it also supports using your stable `get_*()` functions
  example_financial(!!get_id() := letters[1:3])
#> # A tibble: 3 × 2
#>   bank_id other
#>   <chr>   <dbl>
#> 1 a           1
#> 2 b           1
#> 3 c           1
  
  example_financial(!!get_id() := letters[1:3], !!get_europages_product() := 1)
#> # A tibble: 3 × 3
#>   bank_id ep_product other
#>   <chr>        <dbl> <dbl>
#> 1 a                1     1
#> 2 b                1     1
#> 3 c                1     1
```

</details>